### PR TITLE
fix: #538 Variant filter back

### PR DESCRIPTION
**Pull request by francisprovost**
- fix the multi level of dropdown/absolute positioning it add
  - fixes the main issue (back button was closing the popup)
- Small refactoring to help understand the filter components

### CODE REVIEW

File: client/src/components/screens/PatientVariant/components/VariantNavigation.jsx [hunk 1]
```diff
@@ -178,7 +178,7 @@ class VariantNavigation extends React.Component {
     let clickOutside = false;
     if (activeMenu.length > 0) {
       if (activeMenu[0].includes('overflowed-indicator') && !openMenu) openMenu = document.getElementById(`${activeMenu[0]}$Menu`);
-      openOverflowMenu = activeMenu.length > 1 ? document.getElementById(`${activeMenu[0]}$Menu`) : null;
+      openOverflowMenu = activeMenu.length > 1 ? document.getElementById('ant-menu-submenu') : null;
```
Comment (YanLabCHU) [inline client/src/components/screens/PatientVariant/components/VariantNavigation.jsx:181]: Wouldn't it better to use `ref` here?

File: client/src/components/Query/Filter/ApplyButton.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+          { intl.get(`screen.patientvariant.filter.operand.${configOperand}`) }
+        </Menu.Item>
+      )) }
+    </Menu>
+  );
+}
+
+interface Props {
+    config: {
+      operands: any
```
Comment (YanLabCHU) [inline client/src/components/Query/Filter/ApplyButton.tsx:45]: IIRC `operands` is an array of string (`string[]`)

File: client/src/components/Query/Filter/ApplyButton.tsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    },
+    disabled: boolean
+    onClick: () => void,
+    operand: any,
```
Comment (YanLabCHU) [inline client/src/components/Query/Filter/ApplyButton.tsx:49]: `operand` should be a `string`?

**Review by YanLabCHU — CHANGES_REQUESTED** [review]

File: client/src/components/screens/PatientVariant/components/VariantNavigation.jsx (hunk 1 above)
Comment (francisprovost) [reply client/src/components/screens/PatientVariant/components/VariantNavigation.jsx:181]: After a couple of test  it's doesn't seems to be a quick change. I don't think it is worth it to do this change for now

**Review by claudia1296 — APPROVED** [review]
lgtm :)

**Review by YanLabCHU — APPROVED** [review]
LGTM

**Pull request merged**